pygmt/src/text.py

```diff
@@ -157,15 +156,17 @@ def text_(
     # Ensure inputs are either textfiles, x/y/text, or position/text
     if position is None:
         kind = data_kind(textfiles, x, y, text)
+        if kind == "vectors" and text is None:
+            raise GMTInvalidInput("Must provide text with x/y pairs")
     else:
-        if x is not None or y is not None:
+        if x is not None or y is not None or textfiles is not None:
             raise GMTInvalidInput(
-                "Provide either position only, or x/y pairs, not both"
+                "Provide either position only, or x/y pairs, or textfiles."
             )
-        kind = "vectors"
-
-    if kind == "vectors" and text is None:
-        raise GMTInvalidInput("Must provide text with x/y pairs or position")
+        if text is None or is_nonstr_iter(text):
+            raise GMTInvalidInput("Text can't be None or array.")
+        kind = "file"
```

Comment:
It doesn't really matter, but "file" does not seem like an accurate descriptor here to me.
```suggestion
        kind = ""
```